# Assignment 2
by:


## Problem 1

In [1]:
import timeit

In [9]:
def getWords():
    words = []
    with open('Words.txt') as f:
        words = f.readlines()
    return words

In [10]:
def words_letter_position(words, letter, position):
    pos = int(position)
    found = []
    
    for i in range(len(words)):
        temp = words[i]
        temp = temp.strip()
        if len(temp) - 1 >= pos and temp[pos] == letter:
            found.append(temp)
        else:
            continue
    return found

In [11]:
def run(words):
    letter = input('Enter letter: ')
    position = input('Enter position: ')
    w = []   
    start = timeit.default_timer()
    w = words_letter_position(words, letter, position)
    stop = timeit.default_timer()
    return start, stop, w

In [22]:
found_words = []
words = getWords()
start, stop, found_words = run(words)
print('Time: ', stop - start)

Enter letter: b
Enter position: 3
Time:  0.10481770000001234


In [17]:
def getWord2():
    words = ()
    with open('Words.txt') as f:
        words = f.readlines()
    return words

In [23]:
found_words2 = []
words = getWord2()
start, stop, found_words2 = run(words)
print('Time: ', stop - start)

Enter letter: b
Enter position: 3
Time:  0.1476505000000543


### 1-3
Lists are faster than sets
times:
* List: 0.10481770000001234
* Set: 0.1476505000000543

## Problem 2

In [10]:
def readFile():
    words = []
    with open('Words.txt') as f:
        words = f.readlines()
        return words

In [33]:
def anagram(words, wlen, check):
    length_of_word = int(wlen)
    found_anagram = []
    print('1',check)
    
    check = findWord(words, length_of_word, check)
    print('2',check)
    for i in range(len(words)):
        if sorted(words[i]) == sorted(check) and check != words[i]:
            found_anagram.append(words[i])
        return found_anagram

In [46]:
def findWord(words, length, check):
    le = int(length)
    for i in range(len(words)):
        wl = len(words[i])
        if check == '' or check == None and len(words[i]) == le:
            ans = words[i]
            return words[i]
        elif len(words[i]) - 1 == le and sorted(check) and check != words[i]:
            ans = words[i]
            return words[i]
        else:
            continue

In [47]:
x = []
emp = ''
length = input('Enter word length: ')
words = readFile()
check_word = findWord(words, length, emp)
x = anagram(words, length, check_word)

Enter word length: 8
1 ABA

2 abacisci



In [ ]:
for i in range(len(words)):
        temp = words[i]
        temp = temp.strip()
        if len(temp) - 1 == length_of_word and sorted(temp) == sorted(words[i]):
            w = word[i]
            found_anagram.append(words[i])
        return found_anagram